### DISCUSSION TITLE
Tracking cursor in cefview?

### DISCUSSION THREAD

**Discussion by radiaku** [CefView/QCefView#222] [Q&A] [answered]
Hello, how I can tracking mouse in cefview?

windows 10 x64bit
qt5.12
vscpp 2017

setmousetracking to true

adding
```
// connect mouse
  connect(leftCefViewWidget, &CefViewWidget::viewMouseMoved, this, &MainWindow::handleViewMouseEvent);

```

  
CefViewWidget.h
```
signals:
  void viewMouseMoved(QMouseEvent* event); // Declare new signal here

protected slots:
  void onScreenChanged(QScreen* screen);

protected:
  virtual void mouseMoveEvent(QMouseEvent* event) override;

```

still not working, any help?

**Comment by tishion** [comment]
Not possible in non-osr mode

**Reply by radiaku** [reply]
So, do you mean to use OSR mode, set up tracking, connect signals, and then repaint it into the Windows form?

**Comment by tishion** [comment, marked as answer]
you can inherit QCefView and then override this method QCefView::mouseMoveEvent(event);

`
void
CefViewWidget::mouseMoveEvent(QMouseEvent* event)
{
  QCefView::mouseMoveEvent(event);

  // do your stuff here
  qDebug() << "mouse move event:" << event;
}
`

**Reply by radiaku** [reply]
this code in osr mode right?

**Reply by tishion** [reply]
correct
